processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -136,6 +136,7 @@ public GroupByQuery(
         this.limitSpec.build(this.dimensions, this.aggregatorSpecs, this.postAggregatorSpecs);
 
     if (havingSpec != null) {
```

Comment:
I will move this inside `apply` and add a comment about non-thread-safety of HavingSpecs and applyLimit.